.github/scripts/clone_engine_repo

```diff
@@ -2,6 +2,40 @@
 
 set -e
 
+ENGINE_REPO="https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish.git"
+ENGINE="--branch BABEL_2_X_DEV__PG_14_3 $ENGINE_REPO"
+ENGINE_MINOR_VERSION_UPGRADE_FROM="--branch BABEL_2_1_STABLE__PG_14_3 $ENGINE_REPO"
+ENGINE_MAJOR_VERSION_UPGRADE_FROM="--branch BABEL_1_X_DEV__PG_13_6 $ENGINE_REPO"
+PG_HINT_PLAN="--depth 1 --branch REL14_1_4_0 https://github.com/ossc-db/pg_hint_plan.git"
+
+PATTERN=$1
+PR_BODY="$2"
+
+if [ ! "${!PATTERN}" ]; then
+    echo "Invalid pattern: $PATTERN"
+    exit 1
+fi
+
+clone() {
+    echo "running git clone $1"
+
+    exit 0
+    # git clone $1 && exit
+    # rc=$?
+    # echo "git clone failed with return code $rc"
+    # exit $rc
+}
+
+FOUND=$(echo $PR_BODY | grep -o "{$PATTERN:[^\}]*}" | head -1 | sed -n -e "s/{$PATTERN://p" | sed -n -e "s/}$//p")
```

Comment:
The actions themselves have pretty good JSON and regex support, so it might be worth using them for this instead of bash (for that matter, if this clone script gets much more complex we should probably use something other than bash).

I also noticed that the mVU test script (at least in 1.x.dev) is a GH action to do the actual clone, which might well be better than doing it via shell. So perhaps the best thing is to use shell or another language to capture the branch selection logic that we want, but rely on the "checkout@V2" action to do the actual work.